Document Global definitions and different frames (#11)
phlowers/mechaphlowers: docs/user_guide/ug_general_concepts.md

# General concepts

> MechaPhlowers is a library specialized to perform mechanical and geometrical calculus for powerlines. The models are designed to be the simplest and most accurate possible : when possible, calculus are performed in 2D planes.

Reviewer: MechaPhlowers is a library specialized in performing mechanical and geometrical calculations for power lines. The models are designed to be as simple and accurate as possible: whenever appropriate, calculations are performed in 2D planes.
"powerline" collé c'est une marque déposée !

> The granularity is important : the line is composed by several section. For each section, you have several cables. Each cable can be divided into spans, which is the lowest scale of calculus.

Reviewer: Granularity is important: the power line is divided into sections, which are further divided into spans, considered as the smallest units for calculations. Only one phase is studied at a time. A phase consists of a single conductor, a bundle of wires, or an earth wire. Henceforth, the generic term cable will be used to refer to such objects.

[scheme granularity]

> Câble : Une phase peut être composée d’un unique câble ou d’un faisceau de câbles. Dans la suite on parle de câble de manière générique : phase constituée d'un câble simple, phase constituée d'un faisceau de câble, ou câble de garde.

Reviewer: voir au dessus

> The algorithms are switching between the different levels during the resolution

Reviewer: Je ne comprends pas trop cette phrase. Cela veut dire qu'on passe du câble, à la portée, au canton durant les calculs ?
Et si c'est bien cela, c'est vraiment utile de le dire ?

Author: Sentence is deleted

## Definition

![Image not available](./assets/powerline_definitions.drawio.png "Powerlines definitions")

> As you can see below, the study cable is hanging to different suspension strings and start/stopping to a dead end string.

Reviewer: As shown in the figure, the cable under study is suspended from suspension insulators and anchored to dead-end insulators.

> Occasionnaly, some line angles can be found into a section. In this case, we are considering the axe of the pylone (i.e. the arm direction) as the angle bisector.

Reviewer: There may be some angle point in each section. In such cases, the axis of the angle support arm aligns with the bisector of the line's angle.

## Frames

> Depending on the physical models used in the algorithms, the 3D objects, coordinates, forces and moments are projected onto different planes.

Reviewer: Depending on the mathematical models used in the algorithms, 3D objects, coordinates, forces and moments are projected onto different planes.

> The comprehension of those planes are not mandatory to use the package, but needed if you want to dive into the different physical models used.

Reviewer: It isn't necessary to understand those different planes in order to use the package, unless a deeper understanding of the mathematical models is desired.

> However the different frames defined below are needed to display the result : that is why we are defining the transformation operations to go from one to another.

Reviewer: Since the different frames described in the figure are necessary to display the results of the calculations, it is important to explain how to transition from one frame to another.

# General concepts

MechaPhlowers is a library specialized in performing mechanical and geometrical calculations for power lines. The models are designed to be as simple and accurate as possible: whenever appropriate, calculations are performed in 2D planes.

Granularity is important: the power line is divided into sections, which are further divided into spans, considered as the smallest units for calculations. Only one phase is studied at a time. A phase consists of a single conductor, a bundle of wires, or an earth wire. Henceforth, the generic term cable will be used to refer to such objects.

## Definition

> ![Image not available](./assets/powerline_definitions.drawio.png "Powerlines definitions")

Reviewer: ```suggestion
![Image not available](./assets/powerline_definitions.drawio.png "Overhead line definitions")
```

- x and y in the plan, z orthogonal to this plan
- the origin of the frame is the crossing of equator and prime meridian.

!!! important

    For now, only the local frame (in blue in the figure) is considered, with the assumption that the origin is a plane translation relative to point O in the figure.

![Image not available](./assets/earth_frame.drawio.png "Earth frame")

#### Georeferencing

Other coordinate systems may be added in the future.

### Support frame

There are two frames associated with the support:

- The span frame: $\mathcal{R}_{span}$
- The crossarm frame: $\mathcal{R}_{crossarm}$

Those frames are oriented with the z-axis up and the y-axis along the crossarm, pointing inward from the pylon.

![Image not available](./assets/support_frame.drawio.png "Support frame")

!!! important

    - In mechaphlowers, the __crossarm length__ is not the distance between the center and the end of the support, but between the center and the insulator attachment point.
    - The __conductor's attachment altitude__ might not be immediately obvious: it refers to the height of the attachment point above sea level  

### Cable frame / Span frame

> The reference support frame for a span is the left support depending on line direction evolution.

Reviewer: ```suggestion
The reference support frame for a span is the left support depending on support numbering.
```

The cable frame $\mathcal{R}_{cable}$ is defined as described in the figure below. This frame can move depending on the cable loads.

![Image not available](./assets/cable_frame.drawio.png "Cable frame")

> Below a figure to help seeing the cable plane

Reviewer: ```suggestion
The figure below is designed to help understand the cable plane.
```

--8<-- "docs/user_guide/assets/moving_projection_and_cable.html"

### Strategy

> - State changes, unstrained length are performed in the cable plane
> - The equilibrium of moment is performed in the crossarm frame
> - The 2D visualisation "span-oriented" is a projection on the span frame.
> - The 3D visualisation is performed using line frame and could be done with local earth frame too

Reviewer: ```suggestion
- Changes of state and unstrained length calculations are performed in the cable plane
- The equilibrium of moments is performed in the crossarm frame
- The 2D visualisation "span-oriented" is a projection on the span frame.
- The 3D visualisation is performed using the line frame or the local earth frame
```

MechaPhlowers is a library specialized to perform mechanical and geometrical calculus for powerlines. The models are designed to be the simplest and most accurate possible : when possible, calculus are performed in 2D planes.

> The granularity is important : the line is composed by several section. For each section, there are several cables. Each cable can be divided into spans, which is the lowest scale of calculus.

Reviewer: j'ai déjà laissé un commentaire sur l'autre page !
Granularity is important: the power line is divided into sections, which are further divided into spans, considered as the smallest units for calculations. Only one phase is studied at a time. A phase consists of a single conductor, a bundle of wires, or an earth wire. Henceforth, the generic term cable will be used to refer to such objects.

The algorithms are switching between the different levels during the resolution.

## Definition

![Image not available](./assets/powerline_definitions.drawio.png "Powerlines definitions")

> As described below, the study cable is hanging to different suspension strings and start/stopping to a tension support.

Reviewer: As shown in the figure, the cable under study is suspended from suspension insulators and anchored to dead-end insulators.

Occasionnaly, some line angles can be found into a section. In this case, we are considering the axe of the pylone (i.e. the arm direction) as the angle bisector.

## Frames

Depending on the physical models used in the algorithms, the 3D objects, coordinates, forces and moments are projected onto different planes.

Understanding these planes is not required to use the package, but it is necessary if you wish to explore the various physical models employed.

> 

Reviewer: It isn't necessary to understand those different planes in order to use the package, unless a deeper understanding of the mathematical models is desired.

However, the different frames defined below are necessary to display the results. Therefore, we are outlining the transformation operations to transition between them.

> 

Reviewer: Since the different frames described in the figure are necessary to display the results of the calculations, it is important to explain how to transition from one frame to another.

### Earth frame

#### Base frame

The Earth frame  $\mathcal{R}_{earth}$ is defined as the GPS coordinates system :

- x along the west-east axis, facing east
- y along the north-south axis, facing north
- x and y in the plan, z is orthogonal to this plan
- the origin of the frame is the crossing of equator and prime meridian.

!!! important

>     For now, only the local frame is considered (in blue in the figure), with the hypotheses that the origin is a plane translation relative to the O point in the figure.

Reviewer: For now, only the local frame (in blue in the figure) is considered, with the assumption that the origin is a plane translation relative to point O in the figure.

![Image not available](./assets/earth_frame.drawio.png "Earth frame")

#### Georeferencing

> Other coordinates system may be added in the future.

Reviewer: Other coordinate systems may be added in the future.

### Support frame

There are two frames associated with the support:

- The tower body frame: $\mathcal{R}_{towerbody}$
- The crossarm frame: $\mathcal{R}_{crossarm}$

> Those frames are oriented with z axis up and x axis along the crossarm, pointing outwards from the pylon.

Reviewer: Those frames are oriented with the z-axis up and the x-axis along the crossarm, pointing outward from the pylon.

En dehors de cette petite correction, je me demandais, c'est pas x dans le sens de la span ?

Author: As you mentionned we can set x along the span, so y along the arm and pointing inward from the pylone

![Image not available](./assets/support_frame.drawio.png "Support frame")

### Cable frame / Span frame

The reference support frame for a span is the left support depending on line direction evolution.

> The cable frame $\mathcal{R}_{cable}$ is defined as described in the figure below. This frame is then moving depending on cable loads.

Reviewer: This frame can move depending on the cable loads
